zebra/zebra_rib.c

```diff
@@ -2043,7 +2043,11 @@ static void rib_process_result(struct zebra_dplane_ctx *ctx)
 					     dest_str, sizeof(dest_str)));
 		}
 		break;
-	default:
+	case DPLANE_OP_LSP_INSTALL:
```

Comment:
mmm - I'm not sure the rib result handler needs to know about every possible operation (and for example I have more operations coming along...)

Reply:
@donaldsharp suggested this

Reply:
I personally really hate default statements in case statements for enums
the minute we add another enum things just continue to work and you have to rely on the fact that you hit all the code points you have.  I've fixed too many of these issues over the last 2-3 years in FRR to want to continue to program in a paradigm where `default` is ok.  Perhaps we need to have a generic handler that calls the appropriate decode and handle functions based upon the operation?

Reply:
Yes, I think I get it about the 'default' clause. But I don't think every place that wants to see some of the dplane op types needs to be constantly updated as more types arrive. I actually think 'default' is ok here, if we think we need to keep this as a 'switch', because this handler path wants to handle a subset of the operation types, and we expect that some other dispatcher has only called here with that specific subset. I mean, however you get into this handler, if it's constructed as a 'switch', you've either got to use 'default', or open code every known member of the enum (is 'member' the right word?)

Reply:
maybe we need a type/sub-type? 

In this case it would be type == RouteChange
sub-type would be INSTALL/UPDATE/DELETE

Then we could get both worlds of desire here?  As that i also understand your point about having to touch a million places.  I just hate default w/ enum's bugs bugs everywhere